### PR TITLE
Automate script versioning

### PR DESCRIPTION
**Pull request by Kielek**
## Why

I do not want to modify this files manually while releasing.
Upstream changes from https://github.com/signalfx/splunk-otel-dotnet/pull/337

## What

Automate script versioning

## Tests

CI + fork on test

## Checklist

- ~~[ ] `CHANGELOG.md` is updated.~~
- [x] Documentation is updated.
- ~~[ ] New features are covered by tests.~~

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -67,7 +67,7 @@ jobs:
         shell: powershell
         run: |
           $log_path = "C:\ProgramData\OpenTelemetry .NET AutoInstrumentation\logs\*"
-          Import-Module "${PWD}/OpenTelemetry.DotNet.Auto.psm1"
+          Import-Module "${PWD}/bin/installation-scripts/OpenTelemetry.DotNet.Auto.psm1"
```
Comment (pjanotti) [inline .github/workflows/build.yml:70]: Q.: I'm not seeing the symmetrical change changing the location of `otel-dotnet-auto-install.sh`. Are we missing a test for it or is it just tested in a different way?

**Review by pjanotti — APPROVED** [review]
💯 LGTM

File: .github/workflows/build.yml (hunk 1 above)
Comment (nrcventura) [reply .github/workflows/build.yml:70]: It looks like the symmetrical change is in Build.InstallationScripts.Steps.cs. It uses the template scripts to build the installer scripts in the OutputDirectory/installation-scripts directory.

**Review by nrcventura — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (Kielek) [reply .github/workflows/build.yml:70]: `otel-dotnet-auto-install.sh` is tested only post-publish release action https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2984/files#diff-c6d47dacf31662f6791d0cf218b1d34ee0ab48348c037819431cdd40e3fc23e4.

This file is only downloading binaries from GH releases. psm1 skips downloading file from releases in startd ci-build. Full scenario is also covered pos-release activity.

**Pull request merged**